ledger/src/ui/src/bootloader.c

```diff
@@ -0,0 +1,227 @@
+    case RSK_PIN_CMD: // Send pin_buffer
+        reset_if_starting(RSK_META_CMD_UIOP);
+        tx = update_pin_buffer(rx);
+        break;
+    case RSK_IS_ONBOARD: // Wheter it's onboarded or not
+        reset_if_starting(RSK_IS_ONBOARD);
+        tx = is_onboarded();
+        break;
+    case RSK_WIPE: //--- wipe and onboard device ---
+        reset_if_starting(RSK_META_CMD_UIOP);
+        tx = onboard_device(&onboard_ctx);
+        clear_pin();
+        if (mode == BOOTLOADER_MODE_ONBOARD) {
+            *onboard_performed = 1;
+        }
+        break;
+    case RSK_NEWPIN:
+        reset_if_starting(RSK_META_CMD_UIOP);
+        tx = set_pin();
+        clear_pin();
+        break;
+    case RSK_ECHO_CMD: // echo
+        reset_if_starting(RSK_ECHO_CMD);
+        tx = echo(rx);
+        break;
+    case RSK_MODE_CMD: // print mode
+        reset_if_starting(RSK_MODE_CMD);
+        tx = get_mode();
+        break;
+    case INS_ATTESTATION:
+        reset_if_starting(INS_ATTESTATION);
+        tx = get_attestation(rx, &attestation_ctx);
+        break;
+    case INS_SIGNER_AUTHORIZATION:
+        reset_if_starting(INS_SIGNER_AUTHORIZATION);
+        tx = do_authorize_signer(rx, &sigaut_ctx);
+        break;
+    case RSK_RETRIES:
+        reset_if_starting(RSK_RETRIES);
+        tx = get_retries();
+        break;
+    case RSK_UNLOCK_CMD: // Unlock
+        reset_if_starting(RSK_META_CMD_UIOP);
+        tx = unlock();
+        // The pin value will also be used in
+        // BOLOS_UX_CONSENT_APP_ADD command, so we can't wipe the
+        // pin buffer here
+        break;
+    case RSK_END_CMD: // return to dashboard
+        reset_if_starting(RSK_END_CMD);
+        set_autoexec(1);
+        THROW(EX_BOOTLOADER_RSK_END);
+    case RSK_END_CMD_NOSIG: // return to dashboard
+        reset_if_starting(RSK_END_CMD_NOSIG);
+        set_autoexec(0);
+        THROW(EX_BOOTLOADER_RSK_END);
+    default:
+        THROW(ERR_INS_NOT_SUPPORTED);
+        break;
+    }
+
+    return tx;
+}
+
+unsigned int bootloader_process_exception(unsigned short ex, unsigned int tx) {
+    unsigned short sw = 0;
+
+    // Reset the state in case of an error
+    if (ex != APDU_OK) {
+        reset_if_starting(0);
+    }
+
+    switch (ex & 0xF000) {
+    case 0x6000:
+    case 0x9000:
+        sw = ex;
+        break;
+    default:
+        sw = 0x6800 | (ex & 0x7FF);
+        break;
+    }
+
+    // Unexpected exception => report
+    // (check for a potential overflow first)
+    if (tx + 2 > sizeof(G_io_apdu_buffer)) {
+        tx = 0;
+        sw = 0x6983;
+    }
+    SET_APDU_AT(tx++, sw >> 8);
+    SET_APDU_AT(tx++, sw);
+
+    return tx;
+}
+
+// Main function for the bootloader. If mode == BOOTLOADER_MODE_ONBOARD,
+// commands are not accepted after the onboard is performed.
+void bootloader_main(bootloader_mode_t mode) {
+    volatile unsigned int rx = 0;
+    volatile unsigned int tx = 0;
+    volatile unsigned char onboard_performed = 0;
```

Comment:
`onboard_performed` is only used within this module, do you think a `static` global variable (such as e.g. `current_cmd`) should suffice?